core/src/main/scala/org/apache/spark/util/SizeEstimator.scala

```diff
@@ -200,7 +198,7 @@ private[spark] object SizeEstimator extends Logging {
 
       if (length <= ARRAY_SIZE_FOR_SAMPLING) {
         for (i <- 0 until length) {
```

Comment:
while you are doing this, do you mind change the for loop into a while loop? i.e.
```scala
var i = 0
while (i < length) {
  ...
  i += 1
}
```